server/v2/cometbft/abci.go

```diff
@@ -145,6 +155,38 @@ func (c *Consensus[T]) CheckTx(ctx context.Context, req *abciproto.CheckTxReques
 	return cometResp, nil
 }
 
+// checkTxFeeWithMinGasPrices ensure that the provided fees meet a minimum threshold for the validator,
+// if this is a CheckTx. This is only for local mempool purposes, and thus
+// is only ran on check tx.
+func (c *Consensus[T]) checkTxFeeWithMinGasPrices(tx transaction.Tx) error {
+	feeTx, ok := tx.(sdk.FeeTx)
+	if !ok {
+		return errorsmod.Wrap(cometerrors.ErrTxDecode, "tx must be a feeTx")
```

Comment:
We should avoid the error here. Forcing users that don't use our tx to implement an interface specific to our implementation seems like a code smell